src/butil/iobuf.cpp

```diff
@@ -1206,7 +1211,14 @@ int IOBuf::appendv(const const_iovec* vec, size_t n) {
     return 0;
 }
 
-int IOBuf::append_user_data(void* data, size_t size, void (*deleter)(void*)) {
+int IOBuf::append_user_data_with_meta(void* data,
+                                      size_t size,
+                                      void (*deleter)(void*),
+                                      uint64_t meta) {
+    if (size == 0) {
+        LOG(WARNING) << "data_size should not be 0";
+        return -1;
```

Comment:
Did `size == 0` have any side effect? If not, can we return 0 here?

Reply:
This should be unexpected. return -1 is a better choice.

Reply:
@Tuvie Return 0 has been merged into master
https://github.com/apache/incubator-brpc/blob/b788325528d886639e66aa0d46c571c163ec959a/src/butil/iobuf.cpp#L1214-L1220